exercises/flatten-array/flatten-array_spec.lua

```diff
@@ -33,4 +33,11 @@ describe('flatten-array', function()
       flatten({ { { {  } } }, { { } } })
     )
   end)
+
+  it('should flatten array from readme example', function()
+    assert.are.same(
+      {1,2,3,4,5},
+      flatten({1,{2,3,nil,4},{null},5})
```

Comment:
Modifying the readme seemed best -- the second best option is to editorialize in the test spec(though that may not be allowed?)?
In the Tests --> noting that the ability to solve the test case presented as the example is version dependent and therefore more complex than it may seem....  perhaps with a link to the spec/definition in any lua.org of the # operator.  Without that perhaps more pull requests will come?  Wait and see if they do and then react to the additional data points.  I am disappointed that the example.lua does not solve it for the example presented...however - most probably don't realize that. 

regardless I have learned something so I win either way the pull request is dispositioned.   
Thanks